### DISCUSSION TITLE
NetBox 3.x - Unique Tenant name

### DISCUSSION THREAD

**Discussion by moseisleydk** [netbox-community/netbox#17257] [Help Wanted!]
V3.x har builtin support for unique Tenant Group name, but not unique tenant name....

I have some Device Validators as this:

class UniqueSerial(CustomValidator):

    def validate(self, instance):
        from dcim.models import Device

        if instance.serial and Device.objects.exclude(pk=instance.pk).filter(serial=instance.serial).exists():
            self.fail("The Serial number already exists in NetBox", field='serial')

CUSTOM_VALIDATORS = {
    'dcim.device': (
        UniqueSerial()
    )
}

So I would expect this to work:

class UniqueTenantName(CustomValidator):

    def validate(self, instance):
        from tenancy.models import Tenant

        if instance.name and Tenant.objects.exclude(pk=instance.pk).filter(name=instance.name).exists():
            self.fail("The Tenant name already exists in NetBox", field='name')

CUSTOM_VALIDATORS = {
    'dcim.device': (
        UniqueSerial()
    ),
    'tenancy.tenant': (
        UniqueTenantName()
    )
}

------

But I get the error (page):

<class 'TypeError'>

'UniqueTenantName' object is not iterable

Python version: 3.11.4
NetBox version: 3.7.3
Plugins: 
  netbox_secrets: 1.10.2
  netbox_topology_views: 3.9.1

**Comment by AADIP** [comment]
I think that youre missing trailing commas here. Try this:
```
CUSTOM_VALIDATORS = {
    'dcim.device': (UniqueSerial()),
    'tenancy.tenant': (UniqueTenantName()),
}
```
If that doesnt work then you might need commas after your class names.
